python/dolfinx/fem/function.py

```diff
@@ -608,12 +608,21 @@ def tabulate_dof_coordinates(self) -> npt.NDArray[np.float64]:
         return self._cpp_object.tabulate_dof_coordinates()
 
 
-def VectorFunctionSpace(mesh: Mesh, element: typing.Union[ElementMetaData, typing.Tuple[str, int]],
-                        dim=None) -> FunctionSpace:
+def VectorFunctionSpace(
+        mesh: Mesh,
+        element: typing.Union[basix.ufl_wrapper._BasixElementBase,
+                              ElementMetaData, typing.Tuple[str, int]],
+        dim=None) -> FunctionSpace:
     """Create vector finite element (composition of scalar elements) function space."""
-    e = ElementMetaData(*element)
-    ufl_element = basix.ufl_wrapper.create_vector_element(e.family, mesh.ufl_cell().cellname(), e.degree,
-                                                          dim=dim, gdim=mesh.geometry.dim)
+    if isinstance(element, basix.ufl_wrapper._BasixElementBase):
```

Comment:
We try to avoid `isinstance`. Could try/except work here?

Reply:
There's already an `isinstance` in the FunctionSpace constructor that this is copying. I've added a note about removing these to https://github.com/FEniCS/dolfinx/issues/2578, but this should probably be a separate PR

Reply:
Looks like it would be easy enough here. Wouldn't
```python
try:
    ufl_e = basix.ufl_wrapper.create_vector_element(element.family(), element.cell_type, element.degree(),
                                                    element.lagrange_variant, element.dpc_variant, 
                                                    element.discontinuous,
                                                    dim=dim, gdim=mesh.geometry.dim)
except AttributeError:
    ed = ElementMetaData(*element)
    ufl_e = basix.ufl_wrapper.create_vector_element(ed.family, mesh.ufl_cell().cellname(), ed.degree,
                                                    dim=dim, gdim=mesh.geometry.dim)
```
work? (not tested, so not sure about exception type).

Reply:
See #2587.

Reply:
Yes it works, tested with
```python
# FE = basix.ufl_wrapper.BasixElement( something )
V1 = VectorFunctionSpace(mesh, FE, dim=2)
```
and 
```python
V2 = VectorFunctionSpace(mesh, ("P", 2), dim=2)
```
I'll prepare the commit.

Reply:
Looks like `isinstance` is necessary to get mypy pass. I'm not used to mypy, I can't figure out a fix. Step back?

Reply:
See how it’s handled in https://github.com/FEniCS/dolfinx/pull/2587.